rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,446 @@
+
+> This will enable the user to 
+>  * Check if the default Prometheus rules are firing.  
+>  * Check if Alertmanager is able to receive alerts from Prometheus.
+>  * Validate if Alertmanager is able to forward alerts to receivers.
+
+As an additional configuration item, a [`config.yaml`](#installation-yaml) can be passed to the `--alertmanager` flag. This would contain all the configurations required to setup a full feldged Alertmanager install. The documentation for the `config.yaml` file is given [**here**](#installation-yaml).
+
+For the Alertmanager to function properly, it should have the correct iptables and roles, this is done by the proxy injector and RBAC settings. More information about the implementation of the Alertmanager Helm chart can be found in this [**section**](#alertmanager-helm-chart). 
+
+---
+- ## Webhook configuration 
+
+[Webhook.site](https://webhook.site/) is an open source project that allows you to instantly get a unique, random URL that you can use to test and debug Webhooks and HTTP requests. Their API is public, free to use and doesn’t require authentication.  
+
+A unique webhook URL will be generated for the user during the `linkerd install` process and this URL will be configured as the default receiver in the Alertmanager configuration as mentioned in the previous step. 
+
+This involves requesting a token by making a `POST` request to `https://webhook.site/token` with the following parameters.
+```json
+{
+  "default_status": 200,
+  "default_content": "Hello world!",
+  "default_content_type": "text/html",
+  "timeout": 0
+}
+```
+The response contains a **uuid**, which can be used in conjunction with the base url, to access the webhook endpoint at https://webhook.site/{token.uuid}
+```json
+{
+  "redirect": false,
+  "alias": null,
+  "timeout": 0,
+  "premium": true,
+  "uuid": "9981f9f4-657a-4ebf-be7c-1915bedd4775",
+  "ip": "127.0.0.1",
+  "user_agent": "Paw\/3.1.8 (Macintosh; OS X\/10.14.6) GCDHTTPRequest",
+  "default_content": "Hello world!",
+  "default_status": 200,
+  "default_content_type": "text\/plain",
+  "premium_expires_at": "2019-10-22 10:52:20",
+  "created_at": "2019-09-22 10:52:20",
+  "updated_at": "2019-09-22 10:52:20"
+}
+```
+
+An Exmaple of alerts that were triggered from Prometheus (as part of my POC) can be found here [here](https://webhook.site/#!/c21467b9-7ce4-4389-998b-8d91578433f3/ff66d8b1-c206-49a7-b5f8-b44af28a022f/1)
+
+> Privacy can be a concern here as some users may not want their private alerts to go to a public webhook. The default behaviour can be easily overridden using the `config.yaml` file passed to the `--alertmanager` flag 
+
+---
+- ## Installation YAML
+
+The user can pass a yaml file along with the `--alertmanager` flag to override the default settings we provide.
+```sh
+linkerd install --alertmanager config.yaml | kubectl apply -f -
```

Comment:
We already have a configuration model for add-ons right, using `add-on` flag. It is highly suggested to use that instead.

Reply:
Yes noted. I will make the change